delivery/common/jwt/lib/jwt.js

```diff
@@ -0,0 +1,128 @@
+import { crypto } from "crypto";
+
+import { atob, base64url, TextEncoder } from "encoding";
+
+class JWTUtil {
+    static isEmptyString(str) {
+        return !str || 0 === str.trim().length;
+    }
+}
+
+class JWTValidator {
+    constructor(jwtOptions) {
+        this.jwtOptions = jwtOptions || {}, this.algorithms = [ "NONE", "RS256", "RS384", "RS512", "HS256", "HS384", "HS512", "PS256", "PS384", "PS512", "ES256", "ES384", "ES512" ], 
+        this.validateOptionTypes();
+    }
+    async validate(base64JWTToken, keys) {
+        var _a;
+        if ("string" != typeof base64JWTToken) throw new Error("Invalid token type, expected string!");
+        if (!Array.isArray(keys) || !keys.every((elem => void 0 !== elem.type || void 0 !== elem.extractable || null != elem.algorithm || null != elem.usages))) throw new Error("Invalid keys type, expected list of CryptoKey!");
+        const jwtParts = base64JWTToken.split(".");
+        if (jwtParts.length > 3 || jwtParts.length < 2) throw new Error("JWT malformed: Invalid number of parts for JWT token. expected 3 or 2 (unsecured JWT)!");
+        if (JWTUtil.isEmptyString(jwtParts[0])) throw new Error("JWT malformed: jwt header cannot be empty");
+        if (JWTUtil.isEmptyString(jwtParts[1])) throw new Error("JWT malformed: jwt payload cannot be empty");
+        const jwtHBin = atob(jwtParts[0]), jwtPBin = atob(jwtParts[1]), jwtHeader = JSON.parse(jwtHBin), jwtPayload = JSON.parse(jwtPBin);
```

Comment:
Shouldn't the text be treated as base64url? According to https://jwt.io/introduction:

> Then, [the header] JSON is Base64Url encoded to form the first part of the JWT.

and

> The payload is then Base64Url encoded to form the second part of the JSON Web Token.

Reply:
This is changed to base64url decode function.